### DISCUSSION TITLE
yield_fixture to get test TestReport after yield

### DISCUSSION THREAD

**Discussion by parkhomchukp** [pytest-dev/pytest#12180] [Ideas]
In hooks like pytest_runtest_makereport we can easily get the outcome of the test. For example:
```
@pytest.hookimpl(tryfirst=True, hookwrapper=True)
def pytest_runtest_makereport(item, call):
    outcome = yield
    result = outcome.get_result()
```
Then we can do some steps based on the test result.
It would be nice to have such a possibility in yield_fixtures. Code after yield is executed after the test is complete, so seems like it's possible to get a TestReport inside a fixture.
Example:
```
@pytest.yield_fixture(scope='function')
def azure_devops_test_run(azure_devops_test_helper, azure_devops_test_case):
    do_some_stuff()
    outcome = yield
    result = outcome.get_result()
    do_some_other_stuff(result)
```

**Comment by RonnyPfannschmidt** [comment]
Based on the observation that people often ask for this for the wrong reason I request details on the actual use case/intent

**Reply by parkhomchukp** [reply]
I use an external resource to publish my test run results via API. I need to specify the test result in the request I make after every test execution. I know I can use pytest_runtest_makereport hook for this but I need to access some objects that are created at pytest_sessionstart and monkeypatching the session object just feels wrong to me. I could create the session scoped objects I need via fixtures but pytest_runtest_makereport does not have access to fixtures.

**Reply by RonnyPfannschmidt** [reply]
Plugin objects are the right solution
